lessons/07-mcp/README.md

```diff
@@ -0,0 +1,482 @@
+
+**You**: "Yes, Ada sent us. She said you could help us with a protocol?"
+
+**Scipio**: "Protocol? I've spent my life developing strategies for battle. But I suppose a protocol is like a strategy. It helps you organize and manage your resources effectively and ensures communication between them".
+
+Let's talk about flanking maneuvers, infiltration, and the use of cavalry. 
+
+**You**: "Ok, go on?"
+
+**Scipio**: "In battle, we often use flanking maneuvers to attack the enemy from the sides or rear. This disrupts their formations and gives us an advantage. The reason being that the enemy is not prepared for an attack from the side and tends to focus all their attention on the front.  
+
+**You**: "Hm, I guess that's similar to having an app doing too much work up front on a single server. Flanking maneuvers would be like distributing the load across multiple servers, right?"
+
+**Scipio**: "Exactly! By distributing the load, you reduce the risk of a single point of failure and improve overall system resilience. It makes it easier to manage and use resources effectively.
+
+## Model Context Protocol (MCP)
+
+**Time beetle**: "To add to Scipio's point, the Model Context Protocol (MCP) is a way to decentralize your application architecture. It allows you to break down your application into smaller, more manageable components that can operate independently. Here are some important concepts to keep in mind:
+
+- **MCP Hosts**: Programs like IDEs or AI tools that want to access data through MCP.
+- **MCP Clients**: Protocol clients that maintain 1:1 connections with servers.
+- **MCP Servers**: Lightweight programs that expose specific capabilities through MCP.
+- **Local Data Sources**: Files, databases, and services on your computer that MCP servers can securely access.
+- **Remote Services**: External systems available over the internet (e.g., APIs) that MCP servers can connect to.
+
+and here's a quick overview of the MCP architecture:
+
+![Architecture of MCP](https://softchris.github.io/mcp-workshop/assets/images/diagram-791787b0cac57bc331c20e72b645d4f7.png)
+
+### Discovering tools and resources
+
+**You**: "And what about infiltration?"
+
+**Scipio**: "Infiltration is about using small units to infiltrate enemy lines, that way you find out more about the enemy and cause confusion. 
+
+**You**: "Sounds to me like we can use this way of thinking to find out more about our servers, what tools and resources they have.
+
+To sum things up, I should:
+
+- **Use flanking maneuvers**, or in my case break down by app functionality into several MCP servers so that I can distribute them after area of usage and thereby make it easier to scale and manage the app. Such servers can even be updated independently of each other.
```

Comment:
```suggestion
- **Use flanking maneuvers**, or in my case break down features by app functionality into several different MCP servers so that I can distribute servers after area of usage.  Thereby, it's easier to scale and manage the app. Such servers can even be updated independently of each other.
```